openassessment/xblock/staff_area_mixin.py

```diff
@@ -547,17 +547,20 @@ def cancel_submission(self, data, suffix=''):  # pylint: disable=W0613
 
         return self._cancel_workflow(submission_uuid, comments)
 
-    def _cancel_workflow(self, submission_uuid, comments):
+    def _cancel_workflow(self, submission_uuid, comments, requesting_user_id=None):
         """
         Internal helper method to cancel a workflow using the workflow API.
+
+        If requesting_user is not provided, we will use the contextually-current staff user.
         """
         try:
             assessment_requirements = self.workflow_requirements()
-            student_item_dict = self.get_student_item_dict()
+            if requesting_user_id is None:
+                requesting_user_id = self.get_student_item_dict()["student_id"]
```

Comment:
`get_student_item_dict()` gets information about the contextually current user (the user bound to this xblock), which happens to be staff when this is called. I'll try to clarify the comment.